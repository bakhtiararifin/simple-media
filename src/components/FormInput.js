import React from 'react'
import { View } from 'react-native'
import { FormLabel, FormInput, Divider } from 'react-native-elements'

const CustomFormInput = ({ label, ...otherProps }) => {
  return (
    <View>
      <FormLabel>{label}</FormLabel>
      <FormInput {...otherProps} />
      <Divider style={{ marginHorizontal: 15, marginBottom: 15 }} />
    </View>
  )
}

export default CustomFormInput
