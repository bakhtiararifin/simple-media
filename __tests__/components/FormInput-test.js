import React from 'react'
import FormInput from '@components/FormInput'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(<FormInput label="Name" />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly without label', () => {
  const tree = renderer.create(<FormInput />).toJSON()
  expect(tree).toMatchSnapshot()
})
