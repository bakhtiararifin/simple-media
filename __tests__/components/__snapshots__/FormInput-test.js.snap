// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      Name
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e1e8ee",
          "height": 1,
        },
        Object {
          "marginBottom": 15,
          "marginHorizontal": 15,
        },
      ]
    }
  />
</View>
`;

exports[`renders correctly without label 1`] = `
<View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e1e8ee",
          "height": 1,
        },
        Object {
          "marginBottom": 15,
          "marginHorizontal": 15,
        },
      ]
    }
  />
</View>
`;
